airflow/api_connexion/openapi/v1.yaml

```diff
@@ -2169,6 +2235,73 @@ components:
                 $ref: '#/components/schemas/PluginCollectionItem'
         - $ref: '#/components/schemas/CollectionInfo'
 
+    RoleCollectionItem:
+      description: Role collection item
+      type: object
+      properties:
+        name:
+          type: string
+          description: The name of the role
+        actions:
+          type: array
+          items:
+            $ref: '#/components/schemas/ActionResource'
+
+    RoleCollection:
+      description: Role Collections
+      type: object
+      allOf:
+        - type: object
+          properties:
+            roles:
+              type: array
+              items:
+                $ref: '#/components/schemas/RoleCollectionItem'
+        - $ref: '#/components/schemas/CollectionInfo'
+
+    ActionCollectionItem:
+      description: Permission Action Collection Item
+      type: object
+      properties:
+        name:
+          type: string
+          description: The name of the permission
+          nullable: false
+
+    ActionCollection:
+      description: Permission Collection
+      type: object
+      allOf:
+        - type: object
+          properties:
+            actions:
+              type: array
+              items:
+                $ref: '#/components/schemas/ActionCollectionItem'
+        - $ref: '#/components/schemas/CollectionInfo'
+
+    Resource:
+      description: A "resource" on which permissions are granted.
+      type: object
+      properties:
+        name:
+          type: string
+          description: The name of the resource
+          nullable: false
+
+    ActionResource:
+      description: The Action-Resource permission item
```

Comment:
What's the different between a permission and an action resource?

Reply:
ActionResource is equivalent to PermissonView Model https://github.com/dpgaspar/Flask-AppBuilder/blob/55b0976e1450295d5a26a06d28c5b992fb0b561e/flask_appbuilder/security/sqla/models.py#L71
It helps tie action with a resource